src/main/java/de/tum/in/www1/artemis/web/rest/ModelingExerciseResource.java

```diff
@@ -116,25 +111,28 @@ public ModelingExerciseResource(ModelingExerciseRepository modelingExerciseRepos
      * PUT /modeling-exercises : Updates an existing modelingExercise.
      *
      * @param modelingExercise the modelingExercise to update
+     * @param exerciseId the id of the exercise
      * @param notificationText the text shown to students
      * @return the ResponseEntity with status 200 (OK) and with body the updated modelingExercise, or with status 400 (Bad Request) if the modelingExercise is not valid, or with
      *         status 500 (Internal Server Error) if the modelingExercise couldn't be updated
      * @throws URISyntaxException if the Location URI syntax is incorrect
      */
-    @PutMapping("/modeling-exercises")
+    @PutMapping("/modeling-exercises/{exerciseId}")
     @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
-    public ResponseEntity<ModelingExercise> updateModelingExercise(@RequestBody ModelingExercise modelingExercise,
+    public ResponseEntity<ModelingExercise> updateModelingExercise(@PathVariable Long exerciseId, @RequestBody ModelingExercise modelingExercise,
             @RequestParam(value = "notificationText", required = false) String notificationText) throws URISyntaxException {
         log.debug("REST request to update ModelingExercise : {}", modelingExercise);
-        if (modelingExercise.getId() == null) {
-            return createModelingExercise(modelingExercise);
-        }
 
         ResponseEntity<ModelingExercise> responseFailure = checkModelingExercise(modelingExercise);
         if (responseFailure != null) {
             return responseFailure;
         }
 
+        // Check that the modeling exercise id is correct
+        if (!modelingExerciseRepository.existsById(exerciseId) || !exerciseId.equals(modelingExercise.getId())) {
```

Comment:
I think the second condition should have a different error message since that could be pretty confusing otherwise if the exercise actually exists but there is just an id mismatch.

Reply:
You are right I split it into two checks. First is done on the `ModelingExerciseService` and second one here. The second one now says `The updated modeling exercise doesn't match the given id` if failed.